internal/security/client/vaultclient_test.go

```diff
@@ -0,0 +1,252 @@
+			expectedRetry:       true,
+			expectError:         false,
+		},
+		{
+			name:                "New secret client with the same first token again",
+			authToken:           "testToken1",
+			tokenFile:           "testdata/replacement.json",
+			expectedNilCallback: false,
+			expectedNewToken:    "replacement-token",
+			expectedRetry:       true,
+			expectError:         false,
+		},
+		{
+			name:                "New secret client with testToken2, half of TTL remaining",
+			authToken:           "testToken2",
+			expectedNilCallback: false,
+			tokenFile:           "testdata/replacement.json",
+			expectedNewToken:    "replacement-token",
+			expectedRetry:       true,
+			expectError:         false,
+		},
+		{
+			name:                "New secret client with testToken3, less than half of TTL remaining",
+			authToken:           "testToken3",
+			tokenFile:           "testdata/replacement.json",
+			expectedNilCallback: false,
+			expectedNewToken:    "replacement-token",
+			expectedRetry:       true,
+			expectError:         false,
+		},
+		{
+			name:                "New secret client with expired token, no TTL remaining",
+			authToken:           "expiredToken",
+			tokenFile:           "testdata/replacement.json",
+			expectedNilCallback: false,
+			expectedNewToken:    "replacement-token",
+			expectedRetry:       true,
+			expectError:         true,
+		},
+		{
+			name:                "New secret client with expired token, non-existing TokenFile path",
+			authToken:           "expiredToken",
+			tokenFile:           "testdata/non-existing.json",
+			expectedNilCallback: false,
+			expectedNewToken:    "",
+			expectedRetry:       false,
+			expectError:         true,
+		},
+		{
+			name:                "New secret client with expired test token, but same expired replacement token",
+			authToken:           "test-token",
+			tokenFile:           "testdata/testToken.json",
+			expectedNilCallback: false,
+			expectedNewToken:    "test-token",
+			expectedRetry:       false,
+			expectError:         true,
+		},
+		{
+			name:                "New secret client with unauthenticated token",
+			authToken:           "test-token",
+			expectedNilCallback: true,
+			expectedNewToken:    "",
+			expectedRetry:       false,
+			expectError:         true,
+		},
+		{
+			name:                "New secret client with unrenewable token",
+			authToken:           "unrenewableToken",
+			expectedNilCallback: true,
+			expectedNewToken:    "",
+			expectedRetry:       true,
+			expectError:         false,
+		},
+		{
+			name:                "New secret client with no TokenFile",
+			authToken:           "testToken1",
+			tokenFile:           "",
+			expectedNilCallback: true,
+			expectedNewToken:    "",
+			expectedRetry:       false,
+			expectError:         false,
+		},
+	}
+
+	for _, test := range tests {
+		testSecretStoreInfo.TokenFile = test.tokenFile
+		// pinned local variables to avoid scopelint warnings
```

Comment:
pinned? Not familiar with this term.

Reply:
it is like to have it fixed locally here before calling into the `t.Run()` testing func which has go routine inside the implementation of t.Run().   see some example from this link https://www.reddit.com/r/golang/comments/64htjb/i_created_a_linter_tool_for_checking/  if the variables are not pinned.

Reply:
Ok, got it. I haven't seen this done in any table tests... I think we should just start always making a complete copy of test, rather than the fields. i.e:
```
currentTest := test
```
Then use currentTest in the body of the test.

Reply:
can try that :-)